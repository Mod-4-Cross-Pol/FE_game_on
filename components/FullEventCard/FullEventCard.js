import * as React from 'react';
import { StyleSheet, View, TouchableOpacity, Text, Alert } from 'react-native';

export default function FullEventCard({ route }) {
  const { event } = route.params

  const onCountMeInButtonPress = () => {
    Alert.alert('Welcome to the Party! ðŸ¤™')
  }

  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>{event.activity}</Text>

      <View style={styles.lineItem}>
        <Text style={styles.label}>LOCATION:</Text>
        <Text style={styles.text}>{event.location}</Text>
      </View>

      <View style={styles.lineItem}>
        <Text style={styles.label}>START TIME:</Text>
        <Text style={styles.text}>{event.time}</Text>
      </View>

      <View style={styles.lineItem}>
        <Text style={styles.label}>END TIME:</Text>
        <Text style={styles.text}>0000</Text>
      </View>

      <View style={styles.lineItem}>
        <Text style={styles.label}># ATTENDING:</Text>
        <Text style={styles.text}>{event.players_going}</Text>
      </View>

      <View style={styles.lineItem}>
        <Text style={styles.label}># REQUIRED:</Text>
        <Text style={styles.text}>{event.max_players}</Text>
      </View>

      <View style={styles.lineItem}>
        <Text style={styles.label}>SKILL LEVEL:</Text>
        <Text style={styles.text}>{event.skill_level}</Text>
      </View>

      <View style={styles.lineItem}>
      <Text style={styles.label}>EQUIPMENT:</Text>

        <View>
          {event.equipment.map(e => {
            return <Text style={styles.equipmentList}>â€¢ {e.name}</Text>
          })}
        </View>
      </View>

      <TouchableOpacity onPress={onCountMeInButtonPress} style={styles.countMeInButton}><Text style={styles.buttonText}>COUNT ME IN!</Text></TouchableOpacity>
      <Text style={styles.signedUpMessage}>You Are Not Currently</Text>
      <Text style={styles.signedUpMessage}>Signed Up For This Event</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 'auto',
    height: '100%',
    backgroundColor: '#0060b4',
    alignItems: 'center',
    justifyContent: 'center'
  },
  lineItem: {
    justifyContent: 'space-between',
    width: '80%',
    flexDirection: 'row'
  },
  titleText: {
    fontSize: 40,
    fontWeight: 'bold',
    marginBottom: 30,
    color: '#FFF'
  },
  label: {
    fontSize: 20,
    paddingTop: 13
  },
  text: {
    fontSize: 27,
    marginVertical: 5,
    color: '#FFF'
  },
  countMeInButton: {
    backgroundColor: '#FFF',
    height: 50,
    width: '40%',
    borderRadius: 25,
    margin: 15,
    backgroundColor: '#28BFBD',
    borderWidth: 6,
    borderColor: '#dee500',
    justifyContent: 'center',
    alignItems: 'center'
  },
  signedUpMessage: {
    fontSize: 20,
    color: '#dee500'
  },
  equipmentList: {
    fontSize: 20,
    marginVertical: 5,
    color: '#FFF'
  },
  buttonText: {
    fontSize: 18
  }
});
