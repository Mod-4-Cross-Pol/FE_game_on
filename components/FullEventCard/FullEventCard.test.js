import React from 'react';
import { FullEventCard } from './FullEventCard';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('FullEventCard', () => {

  describe('Renders a FullEventCard component', () => {

  let wrapper;

  beforeEach(() => {
    const mockProps = {
      route: {
        params: {
          event: {
            activity: 'Basketball',
            attending: false,
            current_participant_count: 1,
            date: '2020-04-15',
            description: '',
            duration: '45 min',
            equipment: 'fds',
            id: 51,
            lat_long: '37.064236,-95.6431662',
            location: 'Park',
            max_participant_count: 2,
            skill_level: 'Just for fun!',
            start_time: '6:00AM',
          }
        }
      }
    }
      wrapper = Enzyme.shallow(<FullEventCard {...mockProps}/>);
    })

    test('renders the component', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('verfy testing file works', () => {
      expect(true).toBeTruthy();
    });

  });

});
