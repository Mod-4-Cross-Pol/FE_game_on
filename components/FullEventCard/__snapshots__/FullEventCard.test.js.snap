// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullEventCard Renders a FullEventCard component renders the component 1`] = `ShallowWrapper {}`;
